From be6e5f6c126602c499f8104e91c525cd938b4a97 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Thu, 16 Jun 2016 15:21:03 +0200
Subject: [PATCH xserver 7/8] modesetting: Fix swapping of provider sink /
 source capabilities

When a card has import capability it can be an offload _sink_, not
a source and vice versa for export capability.

This commit fixes the modesetting driver to properly set these
capabilities, this went unnoticed sofar because most gpus have both
import and export capability.

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 hw/xfree86/drivers/modesetting/driver.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/xfree86/drivers/modesetting/driver.c b/hw/xfree86/drivers/modesetting/driver.c
index fb05e2e..e7f6e8d 100644
--- a/hw/xfree86/drivers/modesetting/driver.c
+++ b/hw/xfree86/drivers/modesetting/driver.c
@@ -891,11 +891,11 @@ PreInit(ScrnInfoPtr pScrn, int flags)
         if (value & DRM_PRIME_CAP_IMPORT) {
             pScrn->capabilities |= RR_Capability_SinkOutput;
             if (ms->drmmode.glamor)
-                pScrn->capabilities |= RR_Capability_SourceOffload;
+                pScrn->capabilities |= RR_Capability_SinkOffload;
         }
 #if GLAMOR_HAS_GBM_LINEAR
         if (value & DRM_PRIME_CAP_EXPORT && ms->drmmode.glamor)
-            pScrn->capabilities |= RR_Capability_SourceOutput | RR_Capability_SinkOffload;
+            pScrn->capabilities |= RR_Capability_SourceOutput | RR_Capability_SourceOffload;
 #endif
     }
 #endif
-- 
2.7.4

